server/src/main/java/io/crate/execution/engine/sort/Sort.java

```diff
@@ -47,23 +47,25 @@ public static <T> CompletableFuture<List<T>> parallelSort(List<T> list,
             return completedFuture(list);
         }
         List<List<T>> partitions = Lists.partition(list, itemsPerThread);
-        ArrayList<CompletableFuture<List<T>>> futures = new ArrayList<>(partitions.size());
+        CompletableFuture<List<T>>[] futures = new CompletableFuture[partitions.size()];
+        int index = 0;
```

Comment:
`Unchecked assignment: 'java.util.concurrent.CompletableFuture[]' to 'java.util.concurrent.CompletableFuture<java.util.List<T>>[]' `

Again, it's not much of gain here, even less than `list.size`, so maybe let's revert and avoid unsafe code? (but of course account for `List<CompletableFuture> `memory)